client/components/editor.vue

```diff
@@ -244,6 +286,70 @@
       handleWindowResize(e) {
         this.toolbarWidth.width = `${this.$refs.editor.offsetWidth}px`;
       },
+      insertLink() {
+        let {link} = this;
+        if (this.selectedExistingLinks) {
+          // ProseMirror requires for the previous mark to be removed
+          // before adding a new mark
+          this.clearLink();
+          this.selectedExistingLinks = [];
+        }
+        if (link !== '' && this.linkValidationRule(link) === true) {
+          link = link.match(/^[a-zA-Z]+:\/\//) ? link : `http://${link}`;
```

Comment:
Why you have this if you are requiring by regex to URIs to start with `http:`?

Reply:
It's not required by the regex to start with `http`, it just checks if it's present to validate it. So, here, if the URL sent doesn't include a scheme, I add `http` in front of it to avoid treating it like a relative URL.

Reply:
No, I am talking about `linkValidationRule`. That one already validates that URI has a scheme, no?

Reply:
`linkValidationRule` validates if the URL is valid, but it lets the user ignore the scheme `/(https?:\/\/)?/` Then, if the user didn't write the schema, we add it.